### ISSUE TITLE
assigning_clones: suggested code contains expanded macro

### ISSUE THREAD

**Issue by mistydemeo** [rust-lang/rust-clippy#12776]
### Summary

The new `assigning_clones` lint is producing wrong output when the assignment is a macro call. The suggested code is using the expanded macro instead of the macro itself.

### Reproducer

I tried this code:

```rust
let mut s = String::new().clone();
s = format!("{} {}", "hello", "world").to_owned();
```

I expected to see this happen:

A lint output that retains the `format!()` call.

Instead, this happened:

```
warning: assigning the result of `ToOwned::to_owned()` may be inefficient
 --> src/main.rs:3:5
  |
3 |     s = format!("{} {}", "hello", "world").to_owned();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assigning_clones
  = note: `#[warn(clippy::assigning_clones)]` on by default
help: use `clone_into()`
  |
3 ~     {
4 +         let res = $crate::fmt::format($crate::__export::format_args!($($arg)*));
5 +         res
6 ~     }.clone_into(&mut s);
```

### Version

```text
rustc 1.78.0 (9b00956e5 2024-04-29)
binary: rustc
commit-hash: 9b00956e56009bab2aa15d7bff10916599e3d6d6
commit-date: 2024-04-29
host: aarch64-apple-darwin
release: 1.78.0
LLVM version: 18.1.2
```
